### PR TITLE
Create an API GateWay domain name

### PR DESCRIPTION
**Pull request by piuccio**
I've updated aws-sdk dependency because `1.10` doesn't have API GateWay.

The code is a bit rubbish, but that's the best I can do with scala

### CODE REVIEW

File: src/main/scala/com/gu/certificate/Magic.scala [hunk 1]
```diff
@@ -87,13 +89,22 @@ object Magic extends BouncyCastle with FileHelpers {
...
 
-    System.err.println(s"installing to IAM")
-
     val pathPrefix = path.getOrElse("").stripPrefix("/").replace("/", "-")
+    val certificateName = s"$pathPrefix$safeDomain-exp$expDate"
+
+    service match {
+      case Some("iam") => installToIAM(path, region, installCredentialsProvider, certificateName, decryptedPem, certificatePem, chainPem)
+      case Some("apigateway") => installToGateway(region, installCredentialsProvider, safeDomain, certificateName, decryptedPem, certificatePem, chainPem)
+      case _ => None
```
Comment (piuccio) [inline src/main/scala/com/gu/certificate/Magic.scala:101]: this is a bit weird, but at least `case _` is covered by `scopt` input validation.

File: src/main/scala/com/gu/certificate/Magic.scala [hunk 2]
```diff
@@ -109,6 +120,27 @@ object Magic extends BouncyCastle with FileHelpers {
     }
   }
 
+  def installToGateway (region: Region, installCredentialsProvider: AWSCredentialsProvider, safeDomain: String, certificateName: String, decryptedPem: String, certificatePem: String, chainPem: String): Unit = {
+    System.err.println(s"installing to API GateWay")
+
+    val client = region.createClient(classOf[AmazonApiGatewayClient], installCredentialsProvider, null)
+    val createDomainNameRequest = new CreateDomainNameRequest()
+      .withDomainName(safeDomain)
```
Comment (piuccio) [inline src/main/scala/com/gu/certificate/Magic.scala]: I'm not sure if certs should have the same certificate name

**Comment by piuccio** [comment]
I've updated the PR with a bunch of checks

```
install --certificate star.domain.co.uk.crt -p cmsFronts -s apigateway -n banana.theguardian.com

[error] (run-main-1) java.lang.RuntimeException: The API GateWay domain name 'banana.theguardian.com' is not a subdomain of '*.domain.co.uk'
java.lang.RuntimeException: The API GateWay domain name 'banana.theguardian.com' is not a subdomain of '*.domain.co.uk'
```

**Comment by sihil** [comment]
Great stuff. 👍

**Comment by piuccio** [comment]
btw, I don't have push access to this repo, so do you mind merging and deleting the branch for me please?

**Pull request merged**